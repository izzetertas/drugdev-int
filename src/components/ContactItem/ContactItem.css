.wrapper {
  display: flex;
  border: 1px solid #e0d9d9;
  justify-content: space-between !important;
}

@media only screen and (max-width: 400px) {
  .wrapper {
    flex-direction: column;
  }
}

.input {
  max-width: 200px;
  min-width: 100px;
}


.buttons {
  max-width: 120px;
  min-width: 30px;
  display: flex;
  align-content: center;
}