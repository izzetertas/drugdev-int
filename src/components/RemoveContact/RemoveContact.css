.link {
  cursor: pointer;
  margin: 8px;
  width: 20px;
}

.link :hover {
  color: red;
}
