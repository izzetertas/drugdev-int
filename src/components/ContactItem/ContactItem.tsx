import React from 'react'
import ListItem from '@material-ui/core/ListItem';

import RemoveContact from '../RemoveContact';
import Edit from '@material-ui/icons/Edit'
import { Link } from 'react-router-dom';

import './ContactItem.css'

export interface ContactItemProps {
  id: string,
  email: string,
  name: string,
}

export default function ContactItem(props: ContactItemProps){
  return (
    <ListItem className='wrapper' key={props.id}>
      <div className='input'>{props.name}</div>
      <div className='input'>{props.email}</div>
      <div className='buttons'>
        <RemoveContact id={props.id} />
        <Link to={`contacts/${props.id}/edit`} >
          <Edit />
        </Link>
      </div>      
    </ListItem>
  )
}
