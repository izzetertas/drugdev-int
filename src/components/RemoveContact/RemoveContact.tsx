import React, { Fragment } from 'react'
import { Mutation } from 'react-apollo'

import DeleteIcon from '@material-ui/icons/Delete';
import { REMOVE_CONTACT, GET_CONTACTS } from '../queries';

import './RemoveContact.css'

export interface RemoveContactProps {
  id: string,
}

export default function RemoveContact(props: RemoveContactProps) {
  return (
    <Mutation
      mutation={REMOVE_CONTACT}
      variables={{ id: props.id }}
      refetchQueries={[{ query: GET_CONTACTS }]}
    >
      {(removeContact: any) =>
        <DeleteIcon className='link' onClick={removeContact} />
      }
    </Mutation>
  )
}
