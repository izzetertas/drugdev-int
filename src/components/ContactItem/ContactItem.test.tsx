import React from 'react';
import ListItem from '@material-ui/core/ListItem';
import RemoveContact from '../RemoveContact';
import Edit from '@material-ui/icons/Edit'
import { Link } from 'react-router-dom';
import ContactItem from './ContactItem';

describe('<ContactItem />', () => {
  const props = {
    id: 'Asd21asdAGDe',
    email: 'test@mail.com',
    name: 'test',
  }
  let wrapper: any = null
  beforeAll(() => {
    wrapper = shallow(<ContactItem {...props} />);
  })

  it('should render without exploding', () => {
    expect(snapshot(wrapper)).toMatchSnapshot()
  })


  it('should render a ListItem', () => {
    expect(wrapper.find(ListItem)).toHaveLength(1);
  });

  it('should render a RemoveContact', () => {
    const item = wrapper.find(RemoveContact)
    expect(item).toHaveLength(1);
    expect(item.prop('id')).toBe(props.id);
  });

  it('should render a Edit', () => {
    expect(wrapper.find(Edit)).toHaveLength(1);
  });

  it('should render divs correctly', () => {
    expect(wrapper.find('div').first().text()).toBe(props.name);
    expect(wrapper.find('div').at(1).text()).toBe(props.email);
  });
  
  it('should render Link correctly', () => {
    const link = wrapper.find(Link)
    expect(link).toHaveLength(1);
    expect(link.prop('to')).toBe(`contacts/${props.id}/edit`)
  });
});
