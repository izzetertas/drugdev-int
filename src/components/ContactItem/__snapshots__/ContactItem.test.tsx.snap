// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactItem /> should render without exploding 1`] = `
<WithStyles(ListItem)
  className="wrapper"
  key="Asd21asdAGDe"
>
  <div
    className="input"
  >
    test
  </div>
  <div
    className="input"
  >
    test@mail.com
  </div>
  <div
    className="buttons"
  >
    <RemoveContact
      id="Asd21asdAGDe"
    />
    <Link
      to="contacts/Asd21asdAGDe/edit"
    >
      <EditIcon />
    </Link>
  </div>
</WithStyles(ListItem)>
`;
